python/ray/tune/trainable.py

```diff
@@ -440,12 +440,17 @@ def _export_model(self, export_formats, export_dir):
         """
         return {}
 
-
+# TODO(gehring): refactor `wrap_function` and move it to function_runner.py
+#     since the what this function should do is completely dependent on the
+#     how `FunctionRunner` is implemented.
 def wrap_function(train_func):
     from ray.tune.function_runner import FunctionRunner
 
     class WrappedFunc(FunctionRunner):
-        def _trainable_func(self):
-            return train_func
+
+        def _trainable_func(self, config, reporter):
+            output = train_func(config, reporter)
+            reporter(done=True)
+            return output
```

Comment:
In order for this to be compatible with existing trainable functions, I believe `_trainable_func` would need to return a function rather than the output of the called function. Are you proposing an alternative to wrapping functions as Trainable objects for logging?